Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -727,6 +727,12 @@ class Gx2Fitter {
     ACTS_VERBOSE("final params:\n" << params);
     /// Finish Fitting /////////////////////////////////////////////////////////
 
+    if (nUpdate == gx2fOptions.nUpdateMax && gx2fOptions.nUpdateMax > 5) {
```

Comment:
Since currently most of our tracks converge in 4-5 updates, we want to set `nUpdateMax` higher than that to guarantee convergence for most tracks. In cases, where we set a smaller `nUpdateMax`, it's because we want to investigate the behaviour of the fitter before it converges, like in some unit-tests.

Reply:
that might be quite unexpected for the user as this behavior is hidden behind that number without a comment.

maybe add a comment to the option and here why this is done.

I could also imagine that short tracks require less update steps.